flink-clients/src/main/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrap.java

```diff
@@ -282,6 +292,9 @@ private void runApplicationEntryPoint(
             } else {
                 jobIdsFuture.complete(applicationJobIds);
             }
+        } catch (JobStartupFailedException e) {
+            jobIdsFuture.completeExceptionally(e);
```

Comment:
How does this work when there are multiple `execute` calls and we are only failing with `JobStartupFailedException` in the last call but the previous ones succeed?

Reply:
We will basically discard the first job and it will look as if only one job was executed. It follows the principle described here [1]. 
We have to execute the complete user program will all its `execute`s to derive the list of job ids. In case of a failure, the already collected job ids are discarded.

[1] https://github.com/apache/flink/blob/bd4b0aa2ace713ca81ab3aa1ffe3bd86822d365c/flink-clients/src/main/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrap.java#L220